### ISSUE TITLE
Typescript Support

### ISSUE THREAD

**Issue by sbahr001** [hxf31891/react-gradient-color-picker#1]
I was wondering if there is planned support for Typescript type support?

- Thank you so much for this picker.

**Comment by hxf31891** [comment]
Hey @sbahr001, I've been meaning to convert this package to TS for a bit. There are a few bugs I'm cleaning up and then I will add the necessary type checking. Give me a couple weeks.

**Comment by sbahr001** [comment]
Thank you so much!

**Comment by codercatdev** [comment]
Oh this would be amazing!!

**Comment by hxf31891** [comment]
@sbahr001 @codercatdev if either of your have a minute I would greatly appreciate a peak at this:

[ v3 TS compile repo ](https://github.com/hxf31891/react-gradient-color-picker/tree/v3)

As far as I can tell its running well, only pitfall is an increase in package size.

**Comment by jourdanrodrigues** [comment]
TL;DR: let's put up some tests before migrating to TypeScript.

@hxf31891 I'd like to help tackle TypeScript migration. I'm currently doing that at my job and there's a few issues I'm facing that you will too.

The problem here is that there are some spots that have multiple/unknown types going in and out, not to mention some core functionality that's based in side effects (e.g. `gradientParser`). Putting the proper types exposes some issues that need to be addressed because they can cause a crash or unexpected outputs in certain conditions, unless we ignore TypeScript in these spots, which I'm completely against since it's not something urgent.

More than once I had to refactor something in order to achieve proper typing and here's where I'd like to state a prerequisite for migrating to TypeScript: automated tests.

I noticed there's no tests, nor the structure to place some. I already put through a few PRs and was a bit nervous to do so because of the lack of tests. Would be happy to push a PR with a structure to have tests in place, as well as some example (in case it's something you're not familiar with).

I have no knowledge at all of how packages work (e.g. get bundled, delivered, downloaded, etc.), so I can't help with this specific spot.

**Comment by hxf31891** [comment]
@jourdanrodrigues Great plan, I merged your PR and will add some more tests when I get a chance.

**Comment by leidto** [comment]
Any update?

**Comment by shreyas-jadhav** [comment]
🛐
